vyper/compiler/output_bundle.py

```diff
@@ -0,0 +1,232 @@
+import importlib
+import io
+import json
+import os
+import zipfile
+from dataclasses import dataclass
+from functools import cached_property
+from pathlib import PurePath
+from typing import Optional
+
+from vyper.compiler.input_bundle import CompilerInput
+from vyper.compiler.phases import CompilerData
+from vyper.compiler.settings import Settings
+from vyper.semantics.analysis.module import _is_builtin
+
+# data structures and routines for constructing "output bundles",
+# basically reproducible builds of a vyper contract, with varying
+# formats. note this is similar but not exactly analogous to
+# `input_bundle.py` -- the output bundle defined here contains more
+# information.
+
+
+def _anonymize(p: str):
+    segments = []
+    # replace ../../../a/b with 0/1/2/a/b
+    # note that items which "escape" their current package might end up
+    # being invalid paths in the final artifact (they will not resolve
+    # properly during path resolution). TODO sanity check for these
```

Comment:
Follow-up issue?

Reply:
TODO is fine, it's unlikely to happen for most user code